R/lump.R

```diff
@@ -74,14 +74,22 @@ fct_lump <- function(f, n, prop, w = NULL, other_level = "Other",
       rank <- rank(-count, ties = ties.method)
     }
 
-    new_levels <- ifelse(rank <= n, levels, other_level)
+    if(sum(!(rank <= n)) == 1){
```

Comment:
I can't follow the logic here. Could you simplify somehow? I wonder if it might make sense to do an early return of `f`?